client/additional-methods-setup/task.js

```diff
@@ -37,7 +37,7 @@ const createAdditionalMethodsSetupTask = ( {
 		),
 		// please note: marking an item as "dismissed" does not mean it's "completed" - they are considered 2 different things
 		completed: 'yes' === isSetupCompleted,
-		visible: true,
+		visible: false,
```

Comment:
When I removed the task instead of deactivating it, going to `/admin.php?page=wc-admin&task=woocommerce-payments--additional-payment-methods` displayed a blank page. A user goes to this page when:
* Activating UPE from the settings page
* Clicking on "Enable on your store" on the inbox notice on the WCPay overview page and WC home page

These two paths exist even when there's no task, but the page is blank if we remove the task altogether. Mind confirming if this happens on your end?

I evaluated changing the URL for that page so it's not a task but a different screen, but this task is overriding a task promoting the Express checkouts. I didn't go that way because this task only seemed concerned about the duplication, and because of the override this task seems to be doing. I agree this would be neater, but I'm not sure if it's a switch we'd like to do. @LevinMedia what do you think?

Reply:
Sliding in here for a quick comment - yeah, it sucks that in order to see that whole page we _also_ need the task.
Another option (that has been adopter in WCStripe) is to create a new submenu page (via `add_submenu_page()` and `wc_admin_connect_page()`) https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/admin/class-wc-stripe-onboarding-controller.php

This way the task is no longer needed.

Not a blocker or anything, just a note 😄

Reply:
Adding some more context...
For example, enabling UPE from the Settings page. After enabling it, it redirects you to the task URL to select other payment methods https://d.pr/v/nBLTxB
When removing the task (I tested it by commenting out the [task here](https://github.com/Automattic/woocommerce-payments/blob/develop/client/overview/task-list/tasks.js#L135), that screen is empty but we keep redirecting there https://d.pr/v/aSdzW4

If we remove the task, which I think would be neater, looks like we'd need to change the URL used for that screen and take the user there instead after UPE is enabled. I'm all good with going that way, thought on proposing the lowest-impact approach first and going from there.

The behavior is the same when enabling UPE via [this button](https://d.pr/i/TeG7dc) in the inbox notices.
That happens because, from what I noticed, that screen is populated by the task's container property, defined [here.](https://github.com/Automattic/woocommerce-payments/blob/develop/client/additional-methods-setup/task.js#L67).

Maybe there's a way to use the same URL without having to register a task, though, so we'd only need to populate the screen from that URL without changing the URL. Or maybe there's another approach I might have missed.

(I've just noticed @frosso is the other reviewer in this PR. Sorry for the ping @LevinMedia! )

Reply:
@frosso Ha, my bad. I noticed your comment after I sent mine. Mine makes little sense in the thread now.

I've sent this [alternative PR](https://github.com/Automattic/woocommerce-payments/pull/3134) actually removing the task and I'll close this in favor of that one. I'll add Gamma as the reviewer, but please feel free to grab it if either of you wants. Thanks for suggesting to remove the task and for suggesting an approach, @leonardola @frosso